src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -425,11 +439,43 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 return true;
             }
 
-            if (ch == Enter)
+            // If we have a connection info with which to try and re-create the
+            // connection, then let's try doing that on Enter
+            if (_ConnectionInfo != nullptr)

```

Comment:
I guess this puts us in a really weird position, where `ConptyConnection` is responsible for printing out the "press ^D to restart" message, but TermControl is ultimately responsible for handling it.

But at the same time, this means that users of other connections (#4661) won't be told that ^D/Enter are options... but they MIGHT be? If they can be reconstructed via ConnectionInfo?